x-pack/plugins/maps/server/mvt/get_grid_tile.ts

```diff
@@ -59,9 +56,13 @@ export async function getEsGridTile({
       },
       {
         signal: abortController.signal,
+        headers: {
+          'Accept-Encoding': 'gzip',
+        },
+        asStream: true,
       }
     );
-    return tile.body as unknown as Buffer;
+    return await collectStream(tile.body as Iterable<Buffer>);
```

Comment:
Are you doing anything server side with this data? If not, it's more performant to directly pipe the client response body stream into the kibana response stream, as it's much more memory efficient.

Reply:
@nreese made the same suggestion offline. 

The main reason for collecting the stream here to be able to read out the content-length, and return it in the response-header. The previous implementation did this, although may not be necessary agreed.

I'm ignorant of the true impact of just piping the result, so I've put up separate PR with this change (https://github.com/elastic/kibana/pull/122223)
- most of the streams or 1-3 chunks anyway.
- not sure impact on client-side (I'm not sure if TMS allows for streamed http-responses).
- not sure about other Kibana-internals (it's possible that the wrapper-library unpacks the stream anyway)
- not sure about impact on test-coverage

So, to keep this PR smallish, I'd prefer handling the streaming-refactor in isolation (in PR https://github.com/elastic/kibana/pull/122223)